packages/block-editor/src/components/global-styles/hooks.js

```diff
@@ -75,22 +135,51 @@ export function useGlobalSetting( path, blockName, source = 'all' ) {
 	};
 
 	// Unlike styles settings get inherited from top level settings.
-	const resultWithFallback =
-		getSettingValueForContext( blockName ) ?? getSettingValueForContext();
+	const resultWithFallback = useMemo( () => {
+		if ( propertyPath ) {
+			return (
+				getSettingValueForContext( blockName, propertyPath ) ??
+				getSettingValueForContext( undefined, propertyPath )
+			);
+		}
+		const result = {};
+		VALID_SETTINGS.forEach( ( setting ) => {
+			set(
+				result,
+				setting,
+				getSettingValueForContext( blockName, setting ) ??
```

Comment:
I think it would make sense to check if there is any value before applying the set operation to avoid unnecessary set operations and create a smaller resulting object.

Reply:
No, because in that case, we won't be "resetting" values that are undefined.

Reply:
Oh I think I'm misunderstanding what you're proposing here? The only way to get the value is within `getSettingValueForContext` anyway?

Reply:
What I'm proposing is
```
const value = getSettingValueForContext( blockName, setting ) ?? getSettingValueForContext( undefined, setting );
if ( value ) {
set( result, setting, value);
}
```

Wouldn't this work?

Reply:
oh yeah, you're right 👍